src/pytest_inmanta_lsm/remote_service_instance_async.py

```diff
@@ -0,0 +1,544 @@
+        )
+
+        if wait_for_state is not None:
+            # Wait for our service to reach the target state
+            return await self.wait_for_state(
+                target_state=wait_for_state,
+                target_version=wait_for_version,
+                bad_states=bad_states,
+                timeout=timeout,
+                start_version=current_version,
+            )
+        else:
+            return await self.get()
+
+    async def delete(
+        self,
+        *,
+        current_version: typing.Optional[int] = None,
+        wait_for_state: typing.Optional[str] = None,
+        wait_for_version: typing.Optional[int] = None,
+        bad_states: typing.Optional[typing.Collection[str]] = None,
+        timeout: typing.Optional[float] = None,
+    ) -> model.ServiceInstance:
+        """
+        Delete the service instance and wait for it to go into `wait_for_state`.
+
+        :param current_version: current version, defaults to None.
+        :param wait_for_state: wait for this state to be reached, if set to None, returns directly, and doesn't wait.
+        :param wait_for_version: The version the service is expected to be in once we reached the target
+            state.  If we reach this version but not the target state or the opposite, the state will
+            not be a match.
+        :param bad_states: stop waiting and fail if any of these states are reached.  If set to None, defaults to
+            self.UPDATE_FLOW_BAD_STATES.
+        :param timeout: how long can we wait for service to achieve given state (in seconds)
+        :raises BadStateError: If the instance went into a bad state
+        :raises TimeoutError: If the timeout is reached while waiting for the desired state
+        :raises VersionExceededError: If version is provided and the current state goes past it
+        """
+        if current_version is None:
+            current_version = (await self.get()).version
+
+        if bad_states is None:
+            bad_states = self.DELETE_FLOW_BAD_STATES
+
+        LOGGER.info("Deleting service instance %s", self.instance_name)
+        await self.remote_orchestrator.request(
+            "lsm_services_delete",
+            tid=self.remote_orchestrator.environment,
+            service_entity=self.service_entity_name,
+            service_id=self.instance_id,
+            current_version=current_version,
+        )
+
+        if wait_for_state is not None:
+            # Wait for our service to reach the target state
+            return await self.wait_for_state(
+                target_state=wait_for_state,
+                target_version=wait_for_version,
+                bad_states=bad_states,
+                timeout=timeout,
+                start_version=current_version,
+            )
+        else:
+            return await self.get()
+
+    async def set_state(
+        self,
+        state: str,
+        *,
+        current_version: typing.Optional[int] = None,
```

Comment:
`state` and `current_version` are not part of the doc string

Reply:
Indeed, I added them